packages/frontend-2/components/settings/workspaces/security/Discoverability.vue

```diff
@@ -0,0 +1,313 @@
+<template>
+  <section class="py-8">
+    <SettingsSectionHeader title="Workspace discoverability" subheading />
+    <p class="text-body-xs text-foreground-2 mt-2 mb-6">
+      Make it easy for coworkers to join the workspace
+    </p>
+    <CommonCard v-if="workspace?.sso?.provider?.id" class="bg-foundation mb-4">
+      With SSO enabled, allowed domains are configured on your identity provider's side.
+    </CommonCard>
+
+    <div class="flex flex-col space-y-8">
+      <div class="flex">
+        <div class="flex-1 flex-col pr-6 gap-y-1">
+          <p class="text-body-xs font-medium text-foreground">
+            Enable workspace discoverability
+          </p>
+          <p class="text-body-2xs text-foreground-2 leading-5 max-w-md">
+            Lets users discover the workspace if they sign up with a matching email.
+          </p>
+        </div>
+        <div
+          v-tippy="
+            !isWorkspaceAdmin
+              ? 'You must be a workspace admin'
+              : !hasWorkspaceDomains
+              ? 'Your workspace must have at least one verified domain'
+              : undefined
+          "
+        >
+          <FormSwitch
+            v-model="isDomainDiscoverabilityEnabled"
+            name="domain-discoverability"
+            :disabled="!hasWorkspaceDomains || !isWorkspaceAdmin"
+            :show-label="false"
+          />
+        </div>
+      </div>
+
+      <div v-if="isDomainDiscoverabilityEnabled" class="flex flex-col gap-2">
+        <p class="text-body-xs font-medium text-foreground">
+          When someone wants to join
+        </p>
+        <div
+          v-tippy="!isWorkspaceAdmin ? 'You must be a workspace admin' : undefined"
+          :class="!isWorkspaceAdmin ? 'opacity-70' : ''"
```

Comment:
Maybe we should add this disabled/tooltip funcitonality into the FormRadio component if it doesnt have it already